src/Resource/GenericFilesystemResource.php

```diff
@@ -0,0 +1,257 @@
+<?php
+
+/*
+ * This file is part of the puli/repository package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\Repository\Resource;
+
+use Puli\Repository\Api\Resource\FilesystemResource;
+use Puli\Repository\Api\Resource\ResourceMetadata;
+use Puli\Repository\Api\ResourceNotFoundException;
+use Puli\Repository\Api\ResourceRepository;
+use Puli\Repository\Resource\Collection\ArrayResourceCollection;
+
+/**
+ * A generic filesystem resource.
+ *
+ * @since  1.0
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class GenericFilesystemResource implements FilesystemResource
```

Comment:
Yes and no. You can only pass FilesystemResources to add(), but the repository may return either GenericResource or FilesystemResource. If you work with a ResourceRepository (without knowing the implementation) and want to access the filesystem path, you need to do a `instanceof FilesystemResource` check anyway, so that's not a big problem.